### PR TITLE
Feature/detection

### PR DESCRIPTION
**Pull request by afeena**
- Add sqli detection:
 -      check path with libinjection (start subprocess with python2). 
- Move detection and emulation from server script to new script.

### CODE REVIEW

File: base_handler.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            xss_result = self.xss_emulator.handle(session, None, data)
+            if xss_result:
+                detection = {'name': 'xss', 'order': 2, 'payload': xss_result}
+
+        detection = dict(name='unknown', order=0)
+        # dummy for wp-content
+        if re.match(patterns.WORD_PRESS_CONTENT, path):
+            detection = {'name': 'wp-content', 'order': 1}
+
+        sqli = yield from self.check_sqli(path)
```
Comment (glaslos) [inline base_handler.py]: I think we eventually want to pass in the parameter values here: `GET /path?foo=' OR '1'='1' --` as libinjection is evaluating SQL statements.

File: base_handler.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        else:
+            path = urllib.parse.unquote(path)
+            for pattern, patter_details in self.patterns.items():
+                if pattern.match(path):
+                    if detection['order'] < patter_details['order']:
+                        detection = patter_details
+        return detection
+
+    @asyncio.coroutine
+    def emulate(self, detection, session, path):
```
Comment (glaslos) [inline base_handler.py]: With some adjustments, you could do some magic here:
```Python
getattr(self, '{}_emulator'.format(detection['name'])).handle(session, path)
```

**Comment by glaslos** [comment]
I like how you separated the handlers into a separate file to keep the server.py less cluttered.
I added some comments but they are not relevant to the feature so I'm merging this. Feel free to incorporate my comments if you think they make sense.

File: base_handler.py (hunk 2 above)
Comment (glaslos) [inline base_handler.py]: Not sure if it's worth it thou. This makes it harder to understand the code and we only save a couple lines of code and I don't think we are going to add new handlers that often in the future.

File: base_handler.py (hunk 1 above)
Comment (afeena) [inline base_handler.py]: Sorry, I can't clearly understand what did you mean. Can you please explain?

Comment (glaslos) [inline base_handler.py]: Right now you are analyzing this: `/some/path?with=parameter1&and=parameter2` But you actually want to analyze `parameter1` and `parameter2` with libinjection. libinjection evaluates SQL statements, not the entire path with parameters.

**Pull request merged**